var/spack/repos/builtin/packages/libxc/package.py

```diff
@@ -63,31 +67,27 @@ def libs(self):
             libraries, root=self.prefix, shared=shared, recursive=True
         )
 
-    def install(self, spec, prefix):
+    def setup_environment(self, spack_env, run_env):
         # Optimizations for the Intel compiler, suggested by CP2K
         optflags = '-O2'
         if self.compiler.name == 'intel':
-            optflags += ' -xAVX -axCORE-AVX2 -ipo'
+            # Oh flake8 E265, why won't you let us distinguish commented-out
+            # code from discursive comments?
+            #
+            # optflags += ' -xAVX -axCORE-AVX2 -ipo'
+            #
+            # Well, not every lowly login node has AVX or AVX2:
+            #      $ icc  -xAVX -axCORE-AVX2 -ipo hello.c
+            #      $ ./a.out
+            #      Please verify that both the operating system and the \
+            #      processor support Intel(R) AVX instructions.
+            optflags += ' -xSSE4.2 -axCORE-AVX2 -ipo'
```

Comment:
should we add -axAVX too?

Reply:
We can't:

```
$ icc  -xSSE4.2 -xAVX -axCORE-AVX2 -ipo ~/test/c/hello.c 
icc: command line warning #10121: overriding '-xSSE4.2' with '-xAVX'
```

This really should be handled via variants or some other idiom – is there one? I cross-referenced  other packages in commit 5dedb6e .

Reply:
My suggestion was "-axAVX". The "-x" is the default and it should be safe to assume SSE4.2. Each addition "-ax" option will add an optional code path in the executable and a test will be run at the start of the application to see which path should be taken.